dali/ft/filecopy.cpp

```diff
@@ -2520,18 +2528,12 @@ void FileSprayer::setSource(IDistributedFile * source)
         //next.psize = curPart->getDiskSize(true,false);
         sources.append(next);
     }
-
-    gatherFileSizes(false);
-
 }
 
 
 void FileSprayer::setSource(IFileDescriptor * source)
 {
     setSource(source, 0, 1);
-
-    //Now get the size of the files directly (to check they exist).  If they don't exist then switch to the backup instead.
-    gatherFileSizes(false);
```

Comment:
This change is unrelated really and I think removing it may cause problems/changed the behaviour..
This passes false to gatherFileSizes and the method looks at copies and does switches to one of them (that's what the comment is about)..
What happens if primary not found here and mirror copy(ies) used...

I think the gatherFileSize(true) in spray() is a NOP, if this gatherFileSizes(false) call succeeded in getting all sizes, so not a waste to recall.

There may be an issue but it should be tackled in a separate JIRA..
I think what should happen is that it's called once, but that it's reworked to look at all copies and error only if can't find any..

Reply:
Point taken, restored.